addons/barcodes/static/src/js/barcode_events.js

```diff
@@ -0,0 +1,148 @@
+                        keyCode == 32  || // spacebar
+                        (keyCode > 64 && keyCode < 91)   || // letter keys
+                        (keyCode > 95 && keyCode < 112)  || // numpad keys
+                        (keyCode > 185 && keyCode < 193) || // ;=,-./` (in order)
+                        (keyCode > 218 && keyCode < 223)) {   // [\]' (in order))
+                        document.execCommand('insertText', 0, String.fromCharCode(keyCode));
+                    }
+                    $element.trigger({ type: 'keyup', keyCode: keyCode });
+                    send_keys(list);
+                },0);
+            },25);
+        }
+        send_keys(this.buffered_key_events.slice());
+    },
+
+    element_is_editable: function(element) {
+        return $(element).is('input,textarea,[contenteditable="true"]');
+    },
+
+    // This checks that a keypress event is either ESC, TAB or an
+    // arrow key. This is Firefox specific, in Chrom{e,ium}
+    // keypress events are not fired for these types of keys, only
+    // keyup/keydown.
+    is_special_key: function(e) {
+        if (e.key === "ArrowLeft" || e.key === "ArrowRight" ||
+            e.key === "ArrowUp" || e.key === "ArrowDown" ||
+            e.key === "Escape" || e.key === "Tab") {
+            return true;
+        } else {
+            return false;
+        }
+    },
+
+    // The keydown and keyup handlers are here to disallow key
+    // repeat. When preventDefault() is called on a keydown event
+    // the keypress that normally follows is cancelled.
+    keydown_handler: function(e){
+        if (this.key_pressed[e.which]) {
+            e.preventDefault();
+        } else {
+            this.key_pressed[e.which] = true;
+        }
+    },
+
+    keyup_handler: function(e){
+        this.key_pressed[e.which] = false;
+    },
+
+    handler: function(e){
+        if (! this.is_special_key(e)) {
+            // We do not stop events targeting editable elements unless
+            // explicitly asked to by the presence of a barcode_events='true'
+            // attribute because we have no way of redispatching 'genuine' key
+            // events that trigger native event handlers of elements.
+            if (! this.element_is_editable(e.target) || e.target.getAttribute("barcode_events") === "true") {
+                this.buffered_key_events.push(e);
+                e.preventDefault();
+                e.stopImmediatePropagation();
+
+                clearTimeout(this.timeout);
+                if (String.fromCharCode(e.which).match(this.suffix) !== null)
+                    this.handle_buffered_keys();
+                else
+                    this.timeout = setTimeout(this.handle_buffered_keys.bind(this), this.max_time_between_keys_in_ms);
+            }
+        }
+    },
+
+    start: function(prevent_key_repeat){
+        document.body.addEventListener('keypress', this.__handler, true);
+        if (prevent_key_repeat === true) {
```

Comment:
Why not just

```javascript
if (prevent_key_repeat) {
    ....
}
````

It's nicer to read and I don't see why prevent_key_repeat has to be **exactly** true, might as well be something else that evaluates to true.